kanidm_book/src/password_quality.md

```diff
@@ -1,38 +1,52 @@
 # Password Quality and Badlisting
 
-Kanidm embeds a set of tools to help your users use and create strong passwords. 
-This is important as not all user types will require multi-factor authentication (MFA) 
-for their roles, but compromised accounts still pose a risk. There may also be deployment 
+Kanidm embeds a set of tools to help your users use and create strong passwords.
+This is important as not all user types will require multi-factor authentication (MFA)
+for their roles, but compromised accounts still pose a risk. There may also be deployment
 or other barriers to a site rolling out sitewide MFA.
 
 ## Quality Checking
 
-Kanidm enforces that all passwords are checked by the library "[zxcvbn](https://github.com/dropbox/zxcvbn)". 
-This has a large number of checks for password quality. It also provides constructive feedback to users on how 
+Kanidm enforces that all passwords are checked by the library "[zxcvbn](https://github.com/dropbox/zxcvbn)".
+This has a large number of checks for password quality. It also provides constructive feedback to users on how
 to improve their passwords if they are rejected.
 
-Some things that zxcvbn looks for is use of the account name or email in the password, common passwords, 
+Some things that zxcvbn looks for is use of the account name or email in the password, common passwords,
 low entropy passwords, dates, reverse words and more.
 
 This library can not be disabled - all passwords in Kanidm must pass this check.
 
 ## Password Badlisting
 
-This is the process of configuring a list of passwords to exclude from being able to be used. 
-This is especially useful if a specific business has been notified of a compromised account, allowing 
+This is the process of configuring a list of passwords to exclude from being able to be used.
+This is especially useful if a specific business has been notified of compromised accounts, allowing
 you to maintain a list of customised excluded passwords.
 
-The other value to this feature is being able to badlist common passwords that zxcvbn does not detect, or 
+The other value to this feature is being able to badlist common passwords that zxcvbn does not detect, or
 from other large scale password compromises.
 
-By default we ship with a preconfigured badlist that is updated over time as new password breach lists are 
+By default we ship with a preconfigured badlist that is updated over time as new password breach lists are
 made available.
 
-## Updating your own Badlist
+The password badlist by default is append only, meaning it can only grow, but will never remove
```

Comment:
At the moment it's only an ACI blocking the removal. I think I'd need some convincing about "removing" from the badlist, because if an admin says "this password should never be used" why should that be undone to once again allow it to be used?

Reply:
Okay, I think that we *could* have this, but I'd want to hide it on the cli, so we could be temporarily blocked on https://github.com/clap-rs/clap/issues/4353 to add that. But I still am not sure why you'd want to remove a badlisted pw?

Reply:
If it's a performance concern, I tested with 80,000 passwords in the badlist and saw minimal overhead.

Reply:
as we discussed elsewhere this can be a later task, once we can add a hidden command 😄

Reply:
It's possible now, clap looks like it just didn't document it.

Reply:
So... the hidden feature was hidden? Seems a little meta.

Reply:
Rofl. i'm adding the hidden feature now.